server/src-lib/Hasura/GraphQL/Schema.hs

```diff
@@ -583,12 +583,12 @@ buildGCtxMapPG
 buildGCtxMapPG = do
   sc <- askSchemaCache
   gCtxMap <- mkGCtxMap (scTables sc) (scFunctions sc)
-  writeSchemaCache sc {scGCtxMap = gCtxMap}
+  writeSchemaCache sc { scGCtxMap = gCtxMap }
 
-getGCtx :: (CacheRM m) => RoleName -> GCtxMap -> m GCtx
+getGCtx :: (QErrM m) => RoleName -> GCtxMap -> m GCtx
 getGCtx rn ctxMap = do
-  sc <- askSchemaCache
-  return $ fromMaybe (scDefaultRemoteGCtx sc) $ Map.lookup rn ctxMap
+  -- lookup role or return empty
+  pure $ fromMaybe emptyGCtx $ Map.lookup rn ctxMap
```

Comment:
This should throw `500 Internal error` if  `GCtx` for a `RoleName` is not found!

Reply:
That would be a breaking change. This function never threw error. `scDefaultRemoteGCtx sc` can also be an empty value.

Reply:
Another way to think about this being that since `role` is a runtime value, it would seem logical to give an empty GCtx (and not make any asserts) which can throw validation failures later.

Reply:
I agree! Should be this function operate in a monadic context? I don't think we're throwing an exception here.

Reply:
Right. I should remove this from monadic context, the QErr is a remnant from before where I actually threwError and realized it was breaking change :)